cdc/processor/sinkmanager/table_sink_worker.go

```diff
@@ -81,48 +82,218 @@ func (w *sinkWorker) handleTasks(ctx context.Context, taskChan <-chan *sinkTask)
 	}
 }
 
-func (w *sinkWorker) handleTask(ctx context.Context, task *sinkTask) (finalErr error) {
+func validateAndAdjustBound(changefeedID model.ChangeFeedID,
+	task *sinkTask,
+) (engine.Position, engine.Position) {
 	lowerBound := task.lowerBound
-	upperBound := task.getUpperBound(task.tableSink)
+	upperBound := task.getUpperBound(task.tableSink.getReceivedSorterResolvedTs())
+
 	lowerPhs := oracle.GetTimeFromTS(lowerBound.CommitTs)
 	upperPhs := oracle.GetTimeFromTS(upperBound.CommitTs)
-	if upperPhs.Sub(lowerPhs) > maxTaskRange {
-		upperCommitTs := oracle.GoTimeToTS(lowerPhs.Add(maxTaskRange))
-		upperBound = engine.Position{
-			StartTs:  upperCommitTs - 1,
-			CommitTs: upperCommitTs,
-		}
+	// The time range of a task should not exceed maxTaskTimeRange.
+	// This would help for reduce changefeed latency.
+	if upperPhs.Sub(lowerPhs) > maxTaskTimeRange {
+		newUpperCommitTs := oracle.GoTimeToTS(lowerPhs.Add(maxTaskTimeRange))
+		upperBound = engine.GenCommitFence(newUpperCommitTs)
 	}
 
 	if !upperBound.IsCommitFence() {
-		log.Panic("sink task upperbound must be a ResolvedTs",
-			zap.String("namespace", w.changefeedID.Namespace),
-			zap.String("changefeed", w.changefeedID.ID),
+		log.Panic("Table sink task upperbound must be a ResolvedTs",
+			zap.String("namespace", changefeedID.Namespace),
+			zap.String("changefeed", changefeedID.ID),
 			zap.Stringer("span", &task.span),
 			zap.Any("upperBound", upperBound))
 	}
 
+	return lowerBound, upperBound
+}
+
+func needEmitAndAdvance(splitTxn bool, committedTxnSize uint64, pendingTxnSize uint64) bool {
+	// If splitTxn is true, we can safely emit all the events in the last transaction
+	// and current transaction. So we use `committedTxnSize+pendingTxnSize`.
+	splitTxnEmitCondition := splitTxn && committedTxnSize+pendingTxnSize >= maxUpdateIntervalSize
+	// If splitTxn is false, we need to emit the events when the size of the
+	// transaction is greater than maxUpdateIntervalSize.
+	// This could help to reduce the overhead of emit and advance too frequently.
+	noSplitTxnEmitCondition := !splitTxn && committedTxnSize >= maxUpdateIntervalSize
+	return splitTxnEmitCondition ||
+		noSplitTxnEmitCondition
+}
+
+func (w *sinkWorker) doEmitAndAdvance(
```

Comment:
How about
```go
type TableSinkAdvancer struct {
    	events []*model.RowChangedEvent
        currTxnCommitTs uint64,
	lastTxnCommitTs uint64,
	lastPos engine.Position,
	batchID uint64,
	committedTxnSize uint64,
	pendingTxnSize uint64,
}

func (a *TableSinkAdvancer) advance(task *sinkTask, w *worker) {
}
func (a *TableSinkAdvancer) onTaskFinish(task *sinkTask) {
    // handles logic about isLastTime.
}
```

So we can store `TableSinkAdvancer` into `TableSinkWrapper`, so we can do more boundary checks about task range. For example, Task boundaries should be contiguous.

Reply:
Yeap! We can do it in the next PR. Thanks!